TombEngine/Objects/TR5/Entity/tr5_laser_head.cpp

```diff
@@ -1,670 +1,665 @@
-							int v = ((GetRandomControl() & 0x1FFF) + 8192);
-							int c = v * phd_cos(-xRot);
-							target.x = origin.x + c * phd_sin(yRot);
-							target.y = origin.y + v * phd_sin(-xRot);
-							target.z = origin.z + c * phd_cos(yRot);
-
-							if (condition)
-							{
-								LaserHeadData.byte1 = 2;
-								item->TriggerFlags = 0;
-							}
-							else
-								LaserHeadData.byte1 = (GetRandomControl() & 2) + 3;
+						short yRot = 0;
+						if (farAway)
+							yRot = item->Pose.Orientation.y + (GetRandomControl() & 0x3FFF) + ANGLE(135.0f);
+						else
+							yRot = Random::GenerateAngle();
 
-							item->ItemFlags[3] = LaserHeadData.byte1 * ((GetRandomControl() & 3) + 8);
+						int v = Random::GenerateInt(-MAX_VISIBILITY_DISTANCE, MAX_VISIBILITY_DISTANCE);
+						int cosX = v * phd_cos(-xRot);
 
-							LaserHeadData.target.x = target.x;
-							LaserHeadData.target.y = target.y;
-							LaserHeadData.target.z = target.z;
+						target.x = origin.x + cosX * phd_sin(yRot);
+						target.y = origin.y + v * phd_sin(-xRot);
+						target.z = origin.z + cosX * phd_cos(yRot);
+
+						if (farAway)
+						{
+							guardian->trackSpeed = 2;
+							item->TriggerFlags = 0;
 						}
 						else
 						{
-							target.x = LaserHeadData.target.x;
-							target.y = LaserHeadData.target.y;
-							target.z = LaserHeadData.target.z;
+							guardian->trackSpeed = Random::GenerateInt(3, 5);
 						}
 
-						LaserHeadData.byte2 = 0;
-					}
-				}
-				else
-				{
-					LaserHeadData.byte1 = 3;
+						item->ItemFlags[3] = guardian->trackSpeed * ((GetRandomControl() & 3) + 8);
 
-					if (JustLoaded)
-					{
-						int c = 8192 * phd_cos(item->Pose.Orientation.x + 3328);
-
-						target.x = LaserHeadData.target.x = origin.x + c * phd_sin(item->Pose.Orientation.y);
-						target.y = LaserHeadData.target.y = origin.y + SECTOR(8) * phd_sin(3328 - item->Pose.Orientation.x);
-						target.z = LaserHeadData.target.z = origin.z + c * phd_cos(item->Pose.Orientation.y);
+						guardian->target.x = target.x;
+						guardian->target.y = target.y;
+						guardian->target.z = target.z;
 					}
 					else
 					{
-						target.x = LaserHeadData.target.x;
-						target.y = LaserHeadData.target.y;
-						target.z = LaserHeadData.target.z;
+						target.x = guardian->target.x;
+						target.y = guardian->target.y;
+						target.z = guardian->target.z;
 					}
+
+					guardian->trackLara = false;
+				}
+			}
+			else
+			{
+				guardian->trackSpeed = 3;
+
+				if (JustLoaded)
+				{
+					int cosX = phd_cos(item->Pose.Orientation.x + GUARDIAN_ORIENT_OFFSET.x) * ANGLE(45.0f);
+
+					target.x = guardian->target.x = origin.x + cosX * phd_sin(item->Pose.Orientation.y);
+					target.y = guardian->target.y = origin.y + BLOCK(8) * phd_sin(GUARDIAN_ORIENT_OFFSET.x - item->Pose.Orientation.x);
+					target.z = guardian->target.z = origin.z + cosX * phd_cos(item->Pose.Orientation.y);
+				}
+				else
+				{
+					target.x = guardian->target.x;
```

Comment:
Why this condition even exists? `target` local variable is never used afterwards.

Reply:
fixed